cmake/modules/SwiftSupport.cmake

```diff
@@ -0,0 +1,37 @@
+
+# Returns the current achitecture name in a variable
+#
+# Usage:
+#   get_swift_host_arch(result_var_name)
+#
+# If the current architecture is supported by Swift, sets ${result_var_name}
+# with the sanitized host architecture name derived from CMAKE_SYSTEM_PROCESSOR.
+function(get_swift_host_arch result_var_name)
```

Comment:
It feels odd that this would need to live in Yams. Is it possible for this logic to live in Swift or CMake?

Reply:
This is something specific to the layout that is application specific.  This perhaps belongs in some shared swift repository, but, for now, this is replicated.